aiven/resource_project.go

```diff
@@ -160,12 +171,8 @@ func resourceProjectRead(d *schema.ResourceData, m interface{}) error {
 		return err
 	}
 
-	// Don't set card id unconditionally to prevent converting short card id format to long
-	currentCardID, err := getLongCardID(client, d.Get("card_id").(string))
-	if err != nil || currentCardID != project.Card.CardID {
-		if err := d.Set("card_id", project.Card.CardID); err != nil {
-			return err
-		}
+	if err := d.Set("card_id", project.Card.CardID); err != nil {
```

Comment:
Was there a reason for changing this logic? Nothing has changed related to card id handling and need for the special handling still remains afaict.

Reply:
Yep, this is something that I wanted to discuss here. It is a little bit misleading, comment and code itself. We claim we are trying to prevent converting short card id format to long, which is a byproduct of what we are actually doing in the code. We are in fact, only setting `card_id` if long card id has changed or in case of an error (I think we should not set anything in case of an error). I will try to change it a little bit...